.travis.yml

```diff
@@ -0,0 +1,33 @@
+
+language: cpp
+
+addons:
+  apt:
+    sources:
+     - ubuntu-toolchain-r-test
+     - llvm-toolchain-trusty
+    packages:
+     - ninja-build
+     - llvm-dev
+     - libclang-dev
+     - clang
+
+before_install:
+ # Install a supported cmake version (>= 3.4.3)
+ - wget -O cmake.sh https://cmake.org/files/v3.10/cmake-3.10.0-rc1-Linux-x86_64.sh 
+ - sudo sh cmake.sh --skip-license --exclude-subdir --prefix=/usr/local
+
+ # Extract the version number from the most-recently installed LLVM
+ - VERSION=`ls -t /usr/lib/ | grep '^llvm\-' | head -n 1 | sed 's/llvm\-\(.\+\)/\1/'`

```

Comment:
The escaping of `\+` looks wrong here, but apparently it works...? I would guess `\(.+\)` would be a functional capturing group. Likewise, the escaping of `-` is probably not necessary in the sed expression (grep requires it, IIRC).

Reply:
Yeah, I think can simplify it by using the extended regex mode (-E). I never remember which mode escapes what.